src/form_builder/views.py

```diff
@@ -184,7 +184,7 @@ def respond(req, id):
         response_form = ResponseForm(
             req.POST or None, form=user_form, user=req.user)
 
-        if response_form.is_valid() and not user_form.is_closed():
+        if response_form.is_valid() and not user_form.is_closed:
```

Comment:
Minor request, switch the order here:

```py
if not user_form.is_closed and response_form.is_valid():
```

(This will short-circuit.) If the form is closed, there's no need to evaluate the validity of the submission.